### PR TITLE
drop unused images from vhd

### PR DESCRIPTION
**Pull request by haitch**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

- RunCommand image
we don't need runCommand image in VHD, it's a runtime feature, we only need the image upon user request.
missing this image would only block runCommand API request, have no impact on ControlPlane/Node health.
we can save 200MB in the VHD size.

- Kubernetes Dashboard image
Kubernetes dashboard image has no longer been used since 1.18, so drop them.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/components.json [hunk 1]
```diff
@@ -1,15 +1,5 @@
...
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/kubernetes-dashboard:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": []
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/dashboard:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": []
-    },
     {
```
Comment (robbiezhang) [inline vhdbuilder/packer/components.json:3]: you can remove this as well. It's part of the kube-dashboard

**Comment by chmill-zz** [comment]
can we remove the 1.22.* kube-proxy images while we're here. https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/kube-proxy-images.json

**Comment by haitch** [comment]
> can we remove the 1.22.* kube-proxy images while we're here. https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/kube-proxy-images.json  

I don't understand the impact in this case.

**Comment by alexeldeib** [comment]
> I don't understand the impact in this case.

no longer supported, so easy size win

File: vhdbuilder/packer/components.json (hunk 1 above)
Comment (sozercan) [reply vhdbuilder/packer/components.json:3]: +1 on metrics-scaper

**Review by sozercan — APPROVED** [review]

**Review by alexeldeib — APPROVED** [review]

**Review by robbiezhang — APPROVED** [review]

**Pull request merged**